volmdlr/core_compiled.pyx

```diff
@@ -379,8 +379,15 @@ class Vector2D(Vector):
 
     @classmethod
     def remove_duplicate(cls, points):
-        dict_ = {p.approx_hash() : p for p in points}
-        return list(dict_.values())
+        """
+        Don't keep the input list order
+        """
+        input_points = sorted(points, key=lambda pt: (pt.x, pt.y))
+        new_points = [points[0]]
+        for i, point in enumerate(input_points[1:]):
+            if not point.is_close(new_points[-1]):
```

Comment:
For clarity, because it is equivalent.
```suggestion
            if point != new_points[-1]:
```

Reply:
I commented this method because it created some issues, we could search what is the problem